source/common/router/config_impl.cc

```diff
@@ -702,7 +702,8 @@ RouteMatcher::RouteMatcher(const envoy::api::v2::RouteConfiguration& route_confi
   for (const auto& virtual_host_config : route_config.virtual_hosts()) {
     VirtualHostSharedPtr virtual_host(new VirtualHostImpl(virtual_host_config, global_route_config,
                                                           runtime, cm, validate_clusters));
-    for (const std::string& domain : virtual_host_config.domains()) {
+    for (const std::string& domain_name : virtual_host_config.domains()) {
+      const std::string domain = Http::LowerCaseString(domain_name).get();
```

Comment:
As a side note, we should make `Http::LowerCaseString` implementation use https://github.com/envoyproxy/envoy/blob/master/source/common/common/to_lower_table.h instead of `tolower()` which it is currently doing. If you feel like doing that in this PR, go for it. Otherwise don't worry about it.